cmd/fs/helpers_test.go

```diff
@@ -60,3 +69,219 @@ func TestFilerForWindowsLocalPaths(t *testing.T) {
 	testWindowsFilerForPath(t, ctx, `d:\abc`)
 	testWindowsFilerForPath(t, ctx, `f:\abc\ef`)
 }
+
+func mockCurrentUserApi(m *mocks.MockWorkspaceClient, u *iam.User, e error) {
+	currentUserApi := m.GetMockCurrentUserAPI()
+	currentUserApi.EXPECT().Me(mock.AnythingOfType("*context.valueCtx")).Return(u, e)
+}
+
+func mockMustWorkspaceClientFunc(cmd *cobra.Command, args []string) error {
+	return nil
+}
+
+func setupValidArgsFunctionTest(t *testing.T) (*mocks.MockWorkspaceClient, *cobra.Command) {
+	m := mocks.NewMockWorkspaceClient(t)
+	ctx := context.Background()
+	ctx = root.SetWorkspaceClient(ctx, m.WorkspaceClient)
+
+	cmd := &cobra.Command{}
+	cmd.SetContext(ctx)
+
+	return m, cmd
+}
+
+func TestGetValidArgsFunctionCompletion(t *testing.T) {
+	m, cmd := setupValidArgsFunctionTest(t)
+
+	mockCurrentUserApi(m, nil, nil)
+
+	mockFilerForPath := testutil.GetMockFilerForPath(t, "/", []fs.DirEntry{
+		testutil.NewFakeDirEntry("dir", true),
+		testutil.NewFakeDirEntry("file", false),
+	})
+
+	validArgsFunction := getValidArgsFunction(1, false, mockFilerForPath, mockMustWorkspaceClientFunc)
+	completions, directive := validArgsFunction(cmd, []string{}, "dbfs:/")
+
+	// dbfs:/Volumes is programmatically added to the completions
+	assert.Equal(t, []string{"dbfs:/dir", "dbfs:/file", "dbfs:/Volumes"}, completions)
+	assert.Equal(t, cobra.ShellCompDirectiveNoSpace, directive)
+}
+
+func TestGetValidArgsFunctionCompletionOnlyDirs(t *testing.T) {
```

Comment:
We seem to be setting up a mock file system in every test. Could we define a single function that sets up a filer / filesystem that most other tests rely on? It'd be nice to reduce the verbosity of the tests.

Reply:
Removed a few test cases related to manual path joining and added a setupTest method that sets up filer